Marlin/src/lcd/extui/ui_api.h

```diff
@@ -69,7 +69,11 @@ namespace ExtUI {
   enum heater_t   : uint8_t { H0, H1, H2, H3, H4, H5, BED, CHAMBER, COOLER };
   enum fan_t      : uint8_t { FAN0, FAN1, FAN2, FAN3, FAN4, FAN5, FAN6, FAN7 };
   #if HAS_PID_HEATING
-    enum pidresult_t : uint8_t { PID_STARTED, PID_BED_STARTED, PID_CHAMBER_STARTED, PID_BAD_HEATER_ID, PID_TEMP_TOO_HIGH, PID_TUNING_TIMEOUT, PID_DONE };
+    enum pidresult_t : uint8_t {
+      OPTITEM(PIDTEMP, PID_STARTED)
+      OPTITEM(PIDTEMPBED, PID_BED_STARTED)
+      OPTITEM(PIDTEMPCHAMBER, PID_CHAMBER_STARTED)
+      PID_BAD_HEATER_ID, PID_TEMP_TOO_HIGH, PID_TUNING_TIMEOUT, PID_DONE };
```

Comment:
@InsanityAutomation, what are your thoughts on using TERN (indirectly through OPTITEM) to leave items out of an enum based on compilation options?

I'm wondering if this might cause issues if anyone has ExtUI interfaces as a library that is building against Marlin. I know that may have been the original intent for ExtUI, but I don't know whether anyone is doing it in practice.

Reply:
I don't see any gain, as it still occupies the byte. Its needless complexity. 

The event results codes will never be set, handling code should be optimized out by the compiler even if the extui library has handlers for functions not included.

Reply:
Secondary concern... Lets say that enum is passed as a byte or int to a screen to handle locally... Now the bit position for a feature is shifted... It could cause the wrong response to be processed by rearranging the memory underneath it. Imagine using MPC for the hotend, then PID for the bed and bang bang for the chamber then passing the result struct to a remote device.

I don't know that any UI is actively using the byte, however we also dont know that they arnt or wont try to in the future.

Reply:
Your secondary concern is my primary concern.

@classicrocker883, you will need to revert this change to disable portions of the enum in ui_api.h and the various extui switch statements where it was used.

I assume this was just an optimization, and there was no actual issue you were trying to resolve with the change. If reverting this portion of the change will re-introduce some behavior bug, please explain.

Reply:
for instance these switch cases have no default, even before adding `#if - #endif`
```y
        #if ENABLED(PIDTEMP)
          case PID_STARTED:
        #endif
        #if ENABLED(PIDTEMPBED)
          case PID_BED_STARTED:
        #endif
        #if ENABLED(PIDTEMPCHAMBER)
          case PID_CHAMBER_STARTED:
        #endif
 ```
 
 I get the concern for the enum if it were to pass a value that isnt there or shifted, that isnt true in this. I suppose it doesn't matter having the `OPTITEM` or not, and it was just optimization.